flocker/volume/filesystems/memory.py

```diff
@@ -177,8 +178,19 @@ def destroy(self, volume):
     def set_maximum_size(self, volume):
         filesystem = self.get(volume)
         root = filesystem.get_path()
+
+        def get_size():
+            total_size = 0
+            for f in root.walk():
+                if not f.isdir():
+                    total_size += f.getsize()
+            return total_size
+
+        current_size = get_size()
         size_path = root.child(b".size")
         if volume.size.maximum_size is not None:
+            if volume.size.maximum_size < current_size:
+                raise MaximumSizeTooSmall
```

Comment:
Actually not sure why I wrote that line that way. I also have a habit of raising an instance of an exception. So, I will fix this if only for the consistency sake.